tests/ballerina-test-integration/src/test/java/org/ballerinalang/test/packaging/PackagingPushTestCase.java

```diff
@@ -0,0 +1,266 @@
+                                                     + File.separator + "packaging" + File.separator +
+                                                     "Package.md").getAbsolutePath());
+        Files.copy(mdFilePath, mdDir.resolve("Package.md"));
+
+        compressFiles(tempDir, new FileOutputStream(projectPath.resolve(generatedPackagePath)
+                                                               .resolve(packageName + ".zip").toFile()));
+    }
+
+    @Test(description = "Test pushing a package to central")
+    public void testPush() throws Exception {
+        ballerinaClient = new ServerInstance(serverZipPath);
+        String sourceRootPath = projectPath.toString();
+        String[] clientArgs = {"--sourceroot", sourceRootPath, packageName};
+
+        String msg = "integrationtests/" + packageName + ":1.0.0 [project repo -> central]";
+
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        ballerinaClient.addLogLeecher(clientLeecher);
+        ballerinaClient.runMain(clientArgs, getEnvVariables(), "push");
+        clientLeecher.waitForText(5000);
+
+        // Check if package Exists
+        String stagingURL = "https://api.staging-central.ballerina.io/packages/";
+        HttpResponse response = HttpClientRequest.doGet(stagingURL + "integrationtests/" + packageName
+                                                                + "/1.0.0");
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+
+    }
+
+    /**
+     * Get environment variables and add ballerina_home as a env variable the tmp directory.
+     *
+     * @return env directory variable array
+     */
+    private String[] getEnvVariables() {
+        List<String> variables = new ArrayList<>();
+
+        Map<String, String> envVarMap = System.getenv();
+        envVarMap.forEach((key, value) -> variables.add(key + "=" + value));
+        variables.add(ProjectDirConstants.HOME_REPO_ENV_KEY + "=" + tempHomeDirectory.toString());
+        variables.add("BALLERINA_DEV_STAGE_CENTRAL" + "=" + "true");
+
+        return variables.toArray(new String[0]);
+    }
+
+    /**
+     * Generate random package name.
+     *
+     * @param count number of characters required
+     * @return generated name
+     */
+    public String randomPackageName(int count) {
+        String upperCaseAlpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        String lowerCaseAlpha = "abcdefghijklmnopqrstuvwxyz";
+        String alpha = upperCaseAlpha + lowerCaseAlpha;
+        StringBuilder builder = new StringBuilder();
+        while (count-- != 0) {
+            int character = (int) (Math.random() * alpha.length());
+            builder.append(alpha.charAt(character));
+        }
+        return builder.toString();
+    }
+
+    @AfterClass
+    private void cleanup() throws Exception {
+        ballerinaClient.stopServer();
+        Files.walk(tempHomeDirectory)
```

Comment:
move the logic to a function since it's repeated.

Reply:
Fixed with e85d0c1